src/ophyd_async/epics/areadetector/pilatus.py

```diff
@@ -0,0 +1,26 @@
+from ophyd_async.core import StandardDetector, StaticDirectoryProvider
+from ophyd_async.epics.areadetector.drivers.ad_driver import ADDriverShapeProvider
+from ophyd_async.epics.areadetector.writers.hdf_writer import HDFWriter as ADHDFWriter
+from ophyd_async.epics.areadetector.writers.nd_file_hdf import NDFileHDF
+from ophyd_async.epics.areadetector.writers.nd_plugin import NDPluginStats
+
+from .controllers.pilatus_controller import PilatusController
+from .drivers.pilatus_driver import PilatusDriver
+
+dp = StaticDirectoryProvider("/dls/p45/data/cmxxx/i22-yyy", "i22-yyy-")
+
+
+class Pilatus(StandardDetector):
```

Comment:
Here, a class make much more sense to me. I think it's because I can see a factory function like hdf_stats_pilatus and see that it returns something that is a StandardDetector... but I won't know (unless I look into the factory function itself) that this StandardDetector will have a .hdf and a .drv component, for example. In fact, in the original code you have for this, this will be opaque for me anyways because the syntax...

```
def hdf_stats_pilatus(...):
    hdf = ...
    drv = ...
    det = StandardDetector( ..., hdf=hdf, drv=drv)
    return det
```

is kind of opaque. In the background, StandardDetector sets `self.hdf` but just to someone who hasn't looked at the code this is so unreadable. 

I can understand your point of, maybe I shouldn't need to know it has a .drv and a .hdf, but just that it has all the same methods as a StandardDetector, and those are the ones I should use. But, the argument at the end of the day is; who are our users? I know from your perspective it's scientists at beamlines. And from my perspective it also includes people writing and expanding on these devices for example for dodal, who will need to interact and expand on this code beyond the command line.

Reply:
There is a key question here, is CLI access part of our type-safe interface? If so, then we need to use your approach. If not, and it's just for scientists, then we don't. The fundamental part of this is, if you add a new plugin to the chain of an areaDetector, does it become a different sort of object? I would argue no, because the logic is no different, but I can see the argument that it is different, in which case a new class could make sense. 

If we go down the class route then we should be explicit about setting things on self, and also be clear that there are going to be many sorts of Pilatus:

```python
class HDFStatsSumPilatus(StandardDetector):
    def __init__(self, prefix: str, name: str = ""):
        self.drv = PilatusDriver(prefix + "DRV:")
        self.stats = NDPluginStats(prefix + "STATS:"),
        self.hdf = NDFileHDF(prefix + "HDF:")
        super().__init__(
            PilatusController(self.drv),
            ADHDFWriter(
                self.hdf, dp, lambda: self.name, ADDriverShapeProvider(drv), sum="NDStatsSum"
            ),
            config_sigs=[drv.acquire_time, drv.acquire],
            name=name
        )
```

Reply:
After discussion with @coretl and @callumforrester, have agreed to go ahead with classes for now (but name it something more specific like the above)